proposals/lip-0045.md

```diff
@@ -173,10 +195,32 @@ We define the following constants:
 | `EMPTY_BYTES`                                 | bytes  | ""                   | The empty byte string.                                           |
 | `EMPTY_FEE_ADDRESS`                           | bytes  | ""                   | The empty byte string.                                           |
-
-#### uint32be Function
-
-The function `uint32be(x)` returns the big endian uint32 serialization of an integer `x`, with `0 <= x < 2^32`. This serialization is always 4 bytes long.
+| **Length Constants**                          |        |                      |                                                                  |
+| `ADDRESS_LENGTH`                     | uint32  | 20      | Length in bytes of type `Address`. |
+| `ED25519_PUBLIC_KEY_LENGTH`          | uint32  | 32      | Length in bytes of type `PublicKeyEd25519`. |
+| `BLS_PUBLIC_KEY_LENGTH`          | uint32  | 48      | Length in bytes of type `PublicKeyBLS`. |
+| `BLS_SIGNATURE_LENGTH`          | uint32  | 96      | Length in bytes of type `SignatureBLS`. |
+| `MODULE_ID_LENGTH`                   | uint32  | 4       | Length in bytes of type `ModuleID`.|
+| `CROSS_CHAIN_COMMAND_ID_LENGTH`      | uint32  | 2       | Length in bytes of type `ModuleID`.|
+| `CHAIN_ID_LENGTH`                    | uint32  | 4       | Length in bytes of type `ChainID`. |
+| `TOKEN_ID_LENGTH`                    | uint32  | 8       | Length in bytes of a token ID. |
+| `HASH_LENGTH`                 | uint32  | 32      | Length in bytes of outputs of SHA-256 hash function. |
+
+### Type Definition
+
+| Name               | Type    | Validation            | Description                     |
+|--------------------|---------|-----------------------|---------------------------------|
+| `Address` | bytes   | Must be of length `ADDRESS_LENGTH`. | Address of an account.   |
+| `ModuleID` | bytes   | Must be of length `MODULE_ID_LENGTH`. | ID of a module.   |
+| `CrossChainCommandID` | bytes   | Must be of length `CROSS_CHAIN_COMMAND_ID_LENGTH`. | ID of a cross-chain command.   |
+| `ChainID` | bytes   | Must be of length `CHAIN_ID_LENGTH`. | ID of a chain.   |
+| `CCM` | object   | Must follow the `crossChainMessageSchema` schema (defined in [LIP 0049][lip-0049#ccmschema]). | The type of cross-chain messages.   |
+| `CCMID` | bytes   | Must be of length `HASH_LENGTH`. | ID of a cross-chain message.   |
+| `MerkleRoot` | bytes   | Must be of length `HASH_LENGTH`. | The root of a Merkle tree.   |
+| `PublicKeyEd25519` | bytes | Must be of length `ED25519_PUBLIC_KEY_LENGTH`. | Used for Ed25519 public keys. |
+| `PublicKeyBLS` | bytes | Must be of length `BLS_PUBLIC_KEY_LENGTH`. | Used for BLS keys. |
+| `SignatureBLS` | bytes | Must be of length `BLS_SIGNATURE_LENGTH`. | Used for BLS signatures. |
+| `ActiveValidator` | object | Must have `blsKey: PublicKeyBLS` and `bftWeight: uint64` properties. | The BLS key and BFT weight of an active validator. |
 
 ### Interoperability Module Store
```

Comment:
I don't complete understand the distinction between chain and channel, and why three substores are used to store chain-related information under the key `chainID`.

Reply:
Originally all properties were part of a single chain account. Then we decided to split them into those relative to the chain account and those relative to the channel. The outbox root is stored separately only to have shorter inclusion proofs. 

The idea is that chain and channel accounts are updated by seprate properties of the ccu and correspond to different concepts.

Reply:
Thanks for the clarification.